.github/workflows/push_cherrypick.yml

```diff
@@ -0,0 +1,85 @@
+        description: 'Commits to cherry-pick'
+        required: true
+
+env:
+  ACTIONS_PAT: ${{ secrets.ACTIONS_PAT }}
+
+jobs:
+  push-release-branch:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Set up Git
+      run: |
+        git config --global user.name '${{ github.actor }}'
+        git config --global user.email '${{ github.actor }}@users.noreply.github.com'
+        
+    - name: Fetch all branches
+      run: git fetch --all
+
+    - name: Check if branch exists
+      id: check_branch
+      run: |
+        if git show-ref --verify --quiet refs/heads/${{ github.event.inputs.release_branch }}; then
+          echo "Branch already exists: ${{ github.event.inputs.release_branch }}"
+          echo "branch_exists=true" >> $GITHUB_ENV
+        else
+          echo "Branch does not exist: ${{ github.event.inputs.release_branch }}"
+          echo "branch_exists=false" >> $GITHUB_ENV
+        fi
+
+    - name: Create or checkout branch
+      run: |
+        if [ "${{ env.branch_exists }}" = "true" ]; then
+          echo "Checking out existing branch: ${{ github.event.inputs.release_branch }}"
+          git checkout ${{ github.event.inputs.release_branch }}
+        else
+          echo "Creating and checking out new branch: ${{ github.event.inputs.release_branch }}"
+          git checkout -b ${{ github.event.inputs.release_branch }}
+        fi
+
+    - name: Pull latest changes
+      if: env.branch_exists == 'true'
+      run: |
+        echo "Pulling latest changes for branch: ${{ github.event.inputs.release_branch }}"
+        git pull origin ${{ github.event.inputs.release_branch }} --rebase
+
+    - name: Cherry-pick commits
+      id: cherry_pick
+      run: |
+        echo "Cherry-picking commits: ${{ github.event.inputs.cherry_pick_commits }}"
+        git cherry-pick ${{ github.event.inputs.cherry_pick_commits }} || echo "cherry_pick_failed=true" >> $GITHUB_ENV
+
+    - name: Handle cherry-pick result
+      if: env.cherry_pick_failed == 'true'
+      run: |
+        echo "Handling cherry-pick result"
+        if git cherry-pick --continue; then
+          echo "Cherry-pick continued successfully"
+        elif git cherry-pick --skip; then
+          echo "Cherry-pick was skipped"
+        else
+          echo "Cherry-pick encountered an error and was aborted"
+          git cherry-pick --abort
+          exit 1
+        fi
+
+    - name: Push to release branch
+      run: |
+        echo "Setting remote URL with personal access token"
+        git remote set-url origin https://x-access-token:${{ secrets.ACTIONS_PAT }}@github.com/${{ github.repository }}.git
+        echo "Pulling latest changes for branch before pushing"
+        git pull origin ${{ github.event.inputs.release_branch }} --rebase
+        echo "Pushing changes to branch: ${{ github.event.inputs.release_branch }}"
+        git push origin ${{ github.event.inputs.release_branch }}
```

Comment:
Instead of pulling from a branch and then pushing back to it, we want to take the inputs as repo + branch + cherry-pick repo + cherry-pick commit IDs and then push that to a given target repo + branch